### ISSUE TITLE
[Offline support] Support two way offline data sync

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#57]
Blocked by #19, #132, #133.

**Comment by gino-m** [comment]
This task is too broad so closing in favor of more specific steps.

**Comment by gino-m** [comment]
Lots left to do here. Rough plan:

- [ ] Manage projects
  - [ ] Join project - store in local db
  - [ ] Leave project - delete from local db
  - [ ] Activate project - switch projects in local db
  - [ ] Reload project - show message when project config changes w/option to reload or postpone
  - [ ] Reload project - show reload icon next to projects w/pending updates
  - [ ] On activation, store last active project, restore when app opens
- [ ] Offline functionality for features and records
  - [ ] Sync remote feature changes in active projects to local db when app is open
  - [ ] If opening feature while offline, show offline warning
  - [ ] If opening feature when online, download latest records
  - [ ] On remote change reload records for open feature
- [ ] Manage users
  - [ ] Make all data in local db per-user
  - [ ] Allow switching among available accounts in drawer ([docs](https://developer.android.com/training/id-auth)). Fetch and store Firestore auth token on switch.
  - [ ] Add link to manage accounts / add account to system
- [ ] Cleanup / Code Health
  - [ ] Replace *Doc with *Document Maps

**Comment by gino-m** [comment]
Either marked above TODOs as done, or forked into separate issues.

Branch `issue-57/master` has been merged into `master` and deleted. 🚀
